### DISCUSSION TITLE
Expose current AL brightness & color temp level.

### DISCUSSION THREAD

**Discussion by daFreeMan** [basnijholt/adaptive-lighting#398] [Ideas]
It doesn't seem that there a way to get the current brightness level & color temp that AL has calculated should be applied right now. I think that these would be great values to expose to the end user.

For example, I find that in the morning, AL brings up brightness far too quickly for my tastes, especially in the winter when it's still dark outside. Also, now that it's dark early (in the northern hemisphere), it gets too dim too quickly in the evenings. I'd like to be able to work out some sort of calculations of my own to slow down the evening dimming, or possibly turn off AL once it's reached max brightness, then wait until some fixed time in the evening to turn it back on.

All in all, there have been a number of occasions when I'd like to know the brightness level, but haven't been able to figure out where to get it.

Of course, if this is currently available and I've just missed seeing it, _please_ point me in the right direction!

**Comment by mike789** [comment]
For the first part it seems like you are requesting a way to make a custom algorithm for calculating the brightness. Seems like an interesting idea. For now you may want to look at the "max_sunrise_time" and "min_sunset_time" settings while configuring the integration.

As for the second part you can already see the current calculation in the attributes of the main "switch.adaptive_lighting" entity.
![image](https://user-images.githubusercontent.com/2185783/209000298-92e90966-c311-497f-bd72-e18dbe30624c.png)

**Comment by daFreeMan** [comment]
> As for the second part you can already see the current calculation in the attributes of the main "switch.adaptive_lighting" entity.

Interesting, @mike789, I didn't know that. Is it possible to pick those up programmatically? i.e. Have an automation or script read it to set another light, or, perhaps, set the lighting to 50% of what AL says it "should" be when turning on the TV?

Also, can colors actually be set with fractional RGB values as are displayed in your screen shot, or is that just the calculated value and the numbers just truncated/rounded when being used?

**Comment by mike789** [comment]
Yes you can access it using the state_attr() in a template and doing whatever you want with the values.

`{{ state_attr('switch.adaptive_lighting_office','brightness_pct') }}`
`{{ state_attr('switch.adaptive_lighting_office','color_temp_kelvin') }}`

![image](https://user-images.githubusercontent.com/2185783/209166379-d26e946c-e64f-412d-ab68-5c80227065d0.png)

The documentation for the light.turn_on service says that the rgb_color parameter takes integers. But I just tested using floats and it works. Not sure how it deals with them though.

**Comment by basnijholt** [comment]
Floats are automatically converted to integers, so that should work indeed.

**Comment by daFreeMan** [comment]
Oh, that is brilliant! Of course, it was probably pretty obvious to most, but I'm a rank rookie when it comes to figuring out HA. TYVM!!

I'd gotten close, but missed:
`{{ state_attr("switch.adaptive_lighting_adapt_brightness_kitchen", "Kitchen Light Group") }}`
Obviously doesn't work. :(
